src/wp-includes/blocks.php

```diff
@@ -1174,12 +1174,33 @@ function parse_blocks( $content ) {
  */
 function do_blocks( $content ) {
 	$blocks = parse_blocks( $content );
+
+	/**
+	 * Filter that allows plugins to inspect or manipulate the array of parsed
+	 * block objects. The parsed blocks form a hierarchy, rather than a flat
+	 * array, as blocks can be nested within each other during parsing.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array[] $blocks Array of parsed block objects.
+	 */
+	$blocks = apply_filters( 'do_blocks_pre_render', $blocks );
+
 	$output = '';
 
 	foreach ( $blocks as $block ) {
 		$output .= render_block( $block );
 	}
 
+	/**
+	 * Filter to allow plugins to inspect or manipulate the final HTML output.
```

Comment:
Similar to my previous comment, this could become:

```php
* Filters the final HTML output from block rendering.
```

Reply:
Oh, ok. I got you now. I'll replace the reference to the filter with the filter verb 👍

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/5023/commits/7df954bf3eb0f317681e52e7cbaca6c551244244.